admin/class-config.php

```diff
@@ -344,284 +162,195 @@ public function localize_media_script() {
-		} else {
-			$output = $output_input . $output_label;
-		}
+	/**
+	 * Generates the sidebar for admin pages.
+	 *
+	 * @deprecated 1.8.0
+	 */
+	public function admin_sidebar() {
+		_deprecated_function( __METHOD__, 'WPSEO 1.8.0', 'This method is deprecated, please use the <code>Yoast_Form</code> class.' );
 
-		return $output . '<br class="clear" />';
+		Yoast_Form::get_instance()->admin_sidebar();
 	}
 
 	/**
 	 * Create a Text input field.
 	 *
+	 * @deprecated 1.8.0
+	 *
 	 * @param string $var    The variable within the option to create the text input field for.
 	 * @param string $label  The label to show for the variable.
 	 * @param string $option The option the variable belongs to.
-	 *
-	 * @return string
 	 */
 	function textinput( $var, $label, $option = '' ) {
-		if ( empty( $option ) ) {
-			$option = $this->currentoption;
-		}
+		_deprecated_function( __METHOD__, 'WPSEO 1.8.0', 'This method is deprecated, please use the <code>Yoast_Form</code> class.' );
 
-		$options = $this->get_option( $option );
-		$val     = ( isset( $options[ $var ] ) ) ? $options[ $var ] : '';
-
-		return '<label class="textinput" for="' . esc_attr( $var ) . '">' . $label . ':</label><input class="textinput" type="text" id="' . esc_attr( $var ) . '" name="' . esc_attr( $option ) . '[' . esc_attr( $var ) . ']" value="' . esc_attr( $val ) . '"/>' . '<br class="clear" />';
+		if ( $option !== '' ) {
+			Yoast_Form::get_instance()->set_option( $option );
+		}
+		Yoast_Form::get_instance()->textinput( $var, $label );
 	}
 
 	/**
 	 * Create a textarea.
 	 *
+	 * @deprecated 1.8.0
+	 *
 	 * @param string $var    The variable within the option to create the textarea for.
 	 * @param string $label  The label to show for the variable.
 	 * @param string $option The option the variable belongs to.
-	 * @param string $class  The CSS class to assign to the textarea.
-	 *
-	 * @return string
+	 * @param array  $attr   The CSS class to assign to the textarea.
 	 */
-	function textarea( $var, $label, $option = '', $class = '' ) {
-		if ( empty( $option ) ) {
-			$option = $this->currentoption;
-		}
+	function textarea( $var, $label, $option = '', $attr = array() ) {
+		_deprecated_function( __METHOD__, 'WPSEO 1.8.0', 'This method is deprecated, please use the <code>Yoast_Form</code> class.' );
 
-		$options = $this->get_option( $option );
-		$val     = ( isset( $options[ $var ] ) ) ? $options[ $var ] : '';
+		if ( $option !== '' ) {
+			Yoast_Form::get_instance()->set_option( $option );
+		}
 
-		return '<label class="textinput" for="' . esc_attr( $var ) . '">' . esc_html( $label ) . ':</label><textarea class="textinput ' . esc_attr( $class ) . '" id="' . esc_attr( $var ) . '" name="' . esc_attr( $option ) . '[' . esc_attr( $var ) . ']">' . esc_textarea( $val ) . '</textarea>' . '<br class="clear" />';
+		Yoast_Form::get_instance()->textarea( $var, $label, $attr );
```

Comment:
Problem is I changed the arguments, `$option` isn't always in the same spot and it's no longer used in Yoast_Form.